{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Streamlined Extraction of Nucleic Acids and Metabolites from Low- and High-Biomass Samples Using Isopropanol and Matrix Tubes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up notebook environment\n",
    "### NOTE: Use a QIIME2 kernel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import biom\n",
    "import warnings\n",
    "import pickle\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import qiime2 as q2\n",
    "from biom import Table\n",
    "from skbio import OrdinationResults\n",
    "from skbio.stats import subsample_counts\n",
    "from skbio.stats.distance import permanova, anosim, mantel\n",
    "from skbio.stats.distance import DistanceMatrix\n",
    "from qiime2.plugins.gemelli.actions import rpca\n",
    "from qiime2.plugins.gemelli.actions import phylogenetic_rpca_with_taxonomy\n",
    "from qiime2.plugins.feature_table.actions import rarefy\n",
    "from qiime2.plugins.diversity.actions import beta_group_significance\n",
    "from qiime2.plugins.emperor.actions import biplot, plot\n",
    "from qiime2.plugins.diversity.actions import (beta,\n",
    "                                              beta_phylogenetic,\n",
    "                                              pcoa)\n",
    "from qiime2.plugins import demux, deblur, quality_filter, \\\n",
    "                           metadata, feature_table, alignment, \\\n",
    "                           phylogeny, diversity, emperor, feature_classifier, \\\n",
    "                           taxa, composition\n",
    "\n",
    "from assets.step_wise_anova import run_stepwise_anova\n",
    "from qiime2.plugins.fragment_insertion.actions import filter_features\n",
    "warnings.filterwarnings(\"ignore\", category=DeprecationWarning)\n",
    "\n",
    "# helper functions\n",
    "from assets.util_updated_again import (mantel_matched, simulate_depth,\n",
    "                        all_dists, all_dists_no_tree, nested_permanova)\n",
    "\n",
    "# plotting\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "plt.style.use('ggplot')\n",
    "%matplotlib inline\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Subset metadata to make paired files between extraction kits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in sample metadata\n",
    "md = pd.read_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/12201_metadata.txt',\n",
    "                sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset sample metadata to make files for round 1 and round 2\n",
    "md_round1and2 = md[md['round'] != 3]\n",
    "md_round1 = md_round1and2[md_round1and2['round'] == 1]\n",
    "md_round2 = md_round1and2[md_round1and2['round'] == 2]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset round-specific metadata files to make files for each kit\n",
    "md_round1_powersoil = md_round1[md_round1['extraction_kit'] == 'PowerSoil']\n",
    "md_round1_powersoil_pro = md_round1[md_round1['extraction_kit'] == 'PowerSoil Pro']\n",
    "md_round1_norgen = md_round1[md_round1['extraction_kit'] == 'Norgen']\n",
    "md_round2_powersoil = md_round2[md_round2['extraction_kit'] == 'PowerSoil']\n",
    "md_round2_magmax = md_round2[md_round2['extraction_kit'] == 'MagMAX Microbiome']\n",
    "md_round2_nucleomag = md_round2[md_round2['extraction_kit'] == 'NucleoMag Food']\n",
    "md_round2_zymo = md_round2[md_round2['extraction_kit'] == 'Zymo MagBead']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge kit-specific files to make paired files for comparison\n",
    "md_bothPS = pd.concat([md_round1_powersoil, \n",
    "                       md_round1_powersoil_pro,\n",
    "                       md_round1_norgen,\n",
    "                       md_round2_powersoil,\n",
    "                       md_round2_magmax,\n",
    "                       md_round2_nucleomag,\n",
    "                       md_round2_zymo])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export paired files\n",
    "md_bothPS.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1and2_bothPS.txt',\n",
    "                           sep = '\\t',\n",
    "                           index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Stepwise RDA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 16S data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import data\n",
    "md_round1and2_bothPS_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1and2_bothPS.txt')\n",
    "\n",
    "table_16S = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/16S/10_filtered_data/dna_bothPS_16S_deblur_biom_lod_noChl_noMit_sepp_gg_noNTCs_noMock.qza')\n",
    "\n",
    "tree_16S = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/16S/09_fragment_insertion/dna_all_16S_deblur_seqs_noChl_noMit_tree_gg.qza')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Filter table\n",
    "table_16S_biom = table_16S.view(Table)\n",
    "md_round1and2_bothPS_df = md_round1and2_bothPS_q2.to_dataframe()\n",
    "shared_ = list(set(table_16S_biom.ids()) & set(md_round1and2_bothPS_df.index))\n",
    "md_round1and2_bothPS_df_16S = md_round1and2_bothPS_df.reindex(shared_)\n",
    "table_16S_biom_bothPS = table_16S_biom.filter(shared_)\n",
    "keep_ = table_16S_biom_bothPS.ids('observation')[table_16S_biom_bothPS.sum('observation') > 0]\n",
    "table_16S_biom_bothPS.filter(keep_, axis='observation')\n",
    "\n",
    "# Import filtered table and re-indexed metadata file\n",
    "table_16S_bothPS = q2.Artifact.import_data('FeatureTable[Frequency]', table_16S_biom_bothPS)\n",
    "md_round1and2_bothPS_df_16S_q2 = q2.Metadata(md_round1and2_bothPS_df_16S)\n",
    "\n",
    "# Generate distance matrices using 'all_dists' utils\n",
    "rare_depth_16S = 10000\n",
    "dists_res_16S = all_dists(table_16S_bothPS,\n",
    "                      rare_depth_16S, tree_16S)\n",
    "\n",
    "# Generate ordinations (row=samples, cols=axes)\n",
    "pcoa_res_16S = {}\n",
    "pcoa_res_16S['Jaccard'] = pcoa(dists_res_16S['Jaccard'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_16S['Unweighted UniFrac'] = pcoa(dists_res_16S['Unweighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_16S['Weighted UniFrac'] = pcoa(dists_res_16S['Weighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_16S['RPCA'] = dists_res_16S['RPCA'].biplot.view(OrdinationResults).samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp4wdmf1vc/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp4wdmf1vc/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp4wdmf1vc/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total         3.00000    1.00000     \n",
      "Constrained   2.88648    0.96216    3\n",
      "Unconstrained 0.11352    0.03784    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9802 0.9721 0.9343 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.06575 0.02794 0.01984 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp1di8wvgb/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp1di8wvgb/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp1di8wvgb/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total           3.000      1.000     \n",
      "Constrained     2.832      0.944    3\n",
      "Unconstrained   0.168      0.056    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9771 0.9329 0.9219 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.07808 0.06706 0.02287 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp85s5_i1m/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp85s5_i1m/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp85s5_i1m/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total          3.0000     1.0000     \n",
      "Constrained    2.6474     0.8825    3\n",
      "Unconstrained  0.3526     0.1175    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9605 0.9274 0.7595 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.24055 0.07257 0.03952 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp9iveksqu/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp9iveksqu/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmp9iveksqu/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total         3.00000    1.00000     \n",
      "Constrained   2.71206    0.90402    3\n",
      "Unconstrained 0.28794    0.09598    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9336 0.9305 0.8480 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.15204 0.06947 0.06643 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>R2.adj</th>\n",
       "      <th>Df</th>\n",
       "      <th>AIC</th>\n",
       "      <th>F</th>\n",
       "      <th>Pr(&gt;F)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Jaccard</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.955968</td>\n",
       "      <td>60</td>\n",
       "      <td>-1237.612400</td>\n",
       "      <td>232.220113</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.001832</td>\n",
       "      <td>6</td>\n",
       "      <td>-1259.471981</td>\n",
       "      <td>5.188525</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Unweighted UniFrac</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.935939</td>\n",
       "      <td>60</td>\n",
       "      <td>-997.663672</td>\n",
       "      <td>156.598053</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.001607</td>\n",
       "      <td>6</td>\n",
       "      <td>-1008.589714</td>\n",
       "      <td>3.483007</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Weighted UniFrac</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.843865</td>\n",
       "      <td>60</td>\n",
       "      <td>-427.497609</td>\n",
       "      <td>58.560362</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.025052</td>\n",
       "      <td>6</td>\n",
       "      <td>-534.094338</td>\n",
       "      <td>19.442688</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">RPCA</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.888832</td>\n",
       "      <td>60</td>\n",
       "      <td>-644.890708</td>\n",
       "      <td>86.151009</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.004133</td>\n",
       "      <td>6</td>\n",
       "      <td>-663.805165</td>\n",
       "      <td>4.726246</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             R2.adj  Df          AIC  \\\n",
       "Jaccard            + sample_type_3         0.955968  60 -1237.612400   \n",
       "                   + extraction_kit_round  0.001832   6 -1259.471981   \n",
       "Unweighted UniFrac + sample_type_3         0.935939  60  -997.663672   \n",
       "                   + extraction_kit_round  0.001607   6 -1008.589714   \n",
       "Weighted UniFrac   + sample_type_3         0.843865  60  -427.497609   \n",
       "                   + extraction_kit_round  0.025052   6  -534.094338   \n",
       "RPCA               + sample_type_3         0.888832  60  -644.890708   \n",
       "                   + extraction_kit_round  0.004133   6  -663.805165   \n",
       "\n",
       "                                                    F  Pr(>F)  \n",
       "Jaccard            + sample_type_3         232.220113  0.0002  \n",
       "                   + extraction_kit_round    5.188525  0.0002  \n",
       "Unweighted UniFrac + sample_type_3         156.598053  0.0002  \n",
       "                   + extraction_kit_round    3.483007  0.0002  \n",
       "Weighted UniFrac   + sample_type_3          58.560362  0.0002  \n",
       "                   + extraction_kit_round   19.442688  0.0002  \n",
       "RPCA               + sample_type_3          86.151009  0.0002  \n",
       "                   + extraction_kit_round    4.726246  0.0002  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform stepwise RDA-ANOVA\n",
    "es_all = {}\n",
    "use_ = ['round','sample_type','sample_type_2','sample_type_3','biomass_sample','extraction_kit_round']\n",
    "\n",
    "# Clean up meta (only stuff to run)\n",
    "mf_ord = md_round1and2_bothPS_df_16S_q2.to_dataframe().copy()\n",
    "\n",
    "# Filter data\n",
    "keep_ = [v_ for v_ in mf_ord.columns\n",
    "         if len(set(mf_ord[v_])) > 1 and\n",
    "         len(set(mf_ord[v_])) < mf_ord.shape[0]//2]\n",
    "mf_ord = mf_ord[keep_]\n",
    "\n",
    "# Run stepwise ANOVA for all RDA ordinations\n",
    "for metric_, ord_ in  pcoa_res_16S.items():\n",
    "    # get first three axes\n",
    "    ord_ = ord_[[0,1,2]]\n",
    "    ord_.columns = ['PC1','PC2','PC3']\n",
    "    # subset/match\n",
    "    mf_ord_ = mf_ord.copy()\n",
    "    shared_ids = list(set(ord_.index)\\\n",
    "                      & set(mf_ord_.index))\n",
    "    mf_ord_ = mf_ord_.loc[shared_ids,:]\n",
    "    ord_ = ord_.loc[shared_ids,:]\n",
    "    es_all[metric_] = run_stepwise_anova(ord_, mf_ord_, use_) #mf_ord_.columns)\n",
    "\n",
    "# Concat output from all runs and export\n",
    "es_alldf = pd.concat(es_all).rename({'+ sample_type_2':'Sample Type'}, axis=0)\n",
    "es_alldf.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/stepwise_anova/stepwise_anova_16S.txt', sep='\\t')\n",
    "es_alldf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Shotgun data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import data\n",
    "md_round1and2_bothPS_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1and2_bothPS.txt')\n",
    "\n",
    "table_shotgun = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/shotgun/03_filtered_data/dna_bothPS_shotgun_woltka_wol_biom_noNTCs_noMock.qza')\n",
    "\n",
    "tree_shotgun = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/shotgun/wol_tree.qza')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Filter table\n",
    "table_shotgun_biom = table_shotgun.view(Table)\n",
    "md_round1and2_bothPS_df = md_round1and2_bothPS_q2.to_dataframe()\n",
    "shared_ = list(set(table_shotgun_biom.ids()) & set(md_round1and2_bothPS_df.index))\n",
    "md_round1and2_bothPS_df_shotgun = md_round1and2_bothPS_df.reindex(shared_)\n",
    "table_shotgun_biom_bothPS = table_shotgun_biom.filter(shared_)\n",
    "keep_ = table_shotgun_biom_bothPS.ids('observation')[table_shotgun_biom_bothPS.sum('observation') > 0]\n",
    "table_shotgun_biom_bothPS.filter(keep_, axis='observation')\n",
    "\n",
    "# Import filtered table and re-indexed metadata file\n",
    "table_shotgun_bothPS = q2.Artifact.import_data('FeatureTable[Frequency]', table_shotgun_biom_bothPS)\n",
    "md_round1and2_bothPS_df_shotgun_q2 = q2.Metadata(md_round1and2_bothPS_df_shotgun)\n",
    "\n",
    "# Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_shotgun = 2100\n",
    "dists_res_shotgun = all_dists(table_shotgun_bothPS,\n",
    "                      rare_depth_shotgun, tree_shotgun)\n",
    "\n",
    "# Generate ordinations (row=samples, cols=axes)\n",
    "pcoa_res_shotgun = {}\n",
    "pcoa_res_shotgun['Jaccard'] = pcoa(dists_res_shotgun['Jaccard'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_shotgun['Unweighted UniFrac'] = pcoa(dists_res_shotgun['Unweighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_shotgun['Weighted UniFrac'] = pcoa(dists_res_shotgun['Weighted UniFrac'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_shotgun['RPCA'] = dists_res_shotgun['RPCA'].biplot.view(OrdinationResults).samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpz6fxttay/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpz6fxttay/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpz6fxttay/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total         3.00000    1.00000     \n",
      "Constrained   2.82614    0.94205    3\n",
      "Unconstrained 0.17386    0.05795    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9856 0.9643 0.8762 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.12380 0.03571 0.01435 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpp6vv8_e6/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpp6vv8_e6/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpp6vv8_e6/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total         3.00000    1.00000     \n",
      "Constrained   2.79913    0.93304    3\n",
      "Unconstrained 0.20087    0.06696    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9701 0.9555 0.8736 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.12642 0.04454 0.02991 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmplvwedyoh/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmplvwedyoh/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmplvwedyoh/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total         3.00000    1.00000     \n",
      "Constrained   2.70095    0.90032    3\n",
      "Unconstrained 0.29905    0.09968    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9103 0.8980 0.8926 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.10737 0.10199 0.08969 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpypbpze2p/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpypbpze2p/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpypbpze2p/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total          3.0000     1.0000     \n",
      "Constrained    2.6284     0.8761    3\n",
      "Unconstrained  0.3716     0.1239    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9287 0.8753 0.8244 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.17560 0.12466 0.07135 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>R2.adj</th>\n",
       "      <th>Df</th>\n",
       "      <th>AIC</th>\n",
       "      <th>F</th>\n",
       "      <th>Pr(&gt;F)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Jaccard</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.933308</td>\n",
       "      <td>65</td>\n",
       "      <td>-1198.449457</td>\n",
       "      <td>169.578316</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.002959</td>\n",
       "      <td>6</td>\n",
       "      <td>-1228.614727</td>\n",
       "      <td>6.556900</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Unweighted UniFrac</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.921696</td>\n",
       "      <td>65</td>\n",
       "      <td>-1072.603252</td>\n",
       "      <td>142.792139</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.004670</td>\n",
       "      <td>6</td>\n",
       "      <td>-1115.388030</td>\n",
       "      <td>8.588847</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Weighted UniFrac</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.868938</td>\n",
       "      <td>65</td>\n",
       "      <td>-668.788820</td>\n",
       "      <td>80.865936</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.021436</td>\n",
       "      <td>6</td>\n",
       "      <td>-803.389366</td>\n",
       "      <td>24.399744</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">RPCA</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.853082</td>\n",
       "      <td>65</td>\n",
       "      <td>-579.250730</td>\n",
       "      <td>70.946186</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.010697</td>\n",
       "      <td>6</td>\n",
       "      <td>-633.096676</td>\n",
       "      <td>10.396974</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             R2.adj  Df          AIC  \\\n",
       "Jaccard            + sample_type_3         0.933308  65 -1198.449457   \n",
       "                   + extraction_kit_round  0.002959   6 -1228.614727   \n",
       "Unweighted UniFrac + sample_type_3         0.921696  65 -1072.603252   \n",
       "                   + extraction_kit_round  0.004670   6 -1115.388030   \n",
       "Weighted UniFrac   + sample_type_3         0.868938  65  -668.788820   \n",
       "                   + extraction_kit_round  0.021436   6  -803.389366   \n",
       "RPCA               + sample_type_3         0.853082  65  -579.250730   \n",
       "                   + extraction_kit_round  0.010697   6  -633.096676   \n",
       "\n",
       "                                                    F  Pr(>F)  \n",
       "Jaccard            + sample_type_3         169.578316  0.0002  \n",
       "                   + extraction_kit_round    6.556900  0.0002  \n",
       "Unweighted UniFrac + sample_type_3         142.792139  0.0002  \n",
       "                   + extraction_kit_round    8.588847  0.0002  \n",
       "Weighted UniFrac   + sample_type_3          80.865936  0.0002  \n",
       "                   + extraction_kit_round   24.399744  0.0002  \n",
       "RPCA               + sample_type_3          70.946186  0.0002  \n",
       "                   + extraction_kit_round   10.396974  0.0002  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform stepwise RDA-ANOVA\n",
    "es_all = {}\n",
    "use_ = ['round','sample_type','sample_type_2','sample_type_3','biomass_sample','extraction_kit_round']\n",
    "\n",
    "# Clean up meta (only stuff to run)\n",
    "mf_ord = md_round1and2_bothPS_df_shotgun_q2.to_dataframe().copy()\n",
    "\n",
    "# Filter data\n",
    "keep_ = [v_ for v_ in mf_ord.columns\n",
    "         if len(set(mf_ord[v_])) > 1 and\n",
    "         len(set(mf_ord[v_])) < mf_ord.shape[0]//2]\n",
    "mf_ord = mf_ord[keep_]\n",
    "\n",
    "# Run stepwise ANOVA for all RDA ordinations\n",
    "for metric_, ord_ in  pcoa_res_shotgun.items():\n",
    "    # get first three axes\n",
    "    ord_ = ord_[[0,1,2]]\n",
    "    ord_.columns = ['PC1','PC2','PC3']\n",
    "    # subset/match\n",
    "    mf_ord_ = mf_ord.copy()\n",
    "    shared_ids = list(set(ord_.index)\\\n",
    "                      & set(mf_ord_.index))\n",
    "    mf_ord_ = mf_ord_.loc[shared_ids,:]\n",
    "    ord_ = ord_.loc[shared_ids,:]\n",
    "    es_all[metric_] = run_stepwise_anova(ord_, mf_ord_, use_) #mf_ord_.columns)\n",
    "\n",
    "# Concat output from all runs and export\n",
    "es_alldf = pd.concat(es_all).rename({'+ sample_type_2':'Sample Type'}, axis=0)\n",
    "es_alldf.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/stepwise_anova/stepwise_anova_shotgun.txt', sep='\\t')\n",
    "es_alldf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ITS data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import data\n",
    "md_round1and2_bothPS_q2 = q2.Metadata.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/sample_metadata/01_paired_files/12201_metadata_round1and2_bothPS.txt')\n",
    "\n",
    "table_its = q2.Artifact.load('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/data/ITS/08_filtered_data/dna_bothPS_ITS_deblur_biom_lod_noNTCs_noMock.qza')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Filter table\n",
    "table_its_biom = table_its.view(Table)\n",
    "md_round1and2_bothPS_df = md_round1and2_bothPS_q2.to_dataframe()\n",
    "shared_ = list(set(table_its_biom.ids()) & set(md_round1and2_bothPS_df.index))\n",
    "md_round1and2_bothPS_df_its = md_round1and2_bothPS_df.reindex(shared_)\n",
    "table_its_biom_bothPS = table_its_biom.filter(shared_)\n",
    "keep_ = table_its_biom_bothPS.ids('observation')[table_its_biom_bothPS.sum('observation') > 0]\n",
    "table_its_biom_bothPS.filter(keep_, axis='observation')\n",
    "\n",
    "# Import filtered table and re-indexed metadata file\n",
    "table_its_bothPS = q2.Artifact.import_data('FeatureTable[Frequency]', table_its_biom_bothPS)\n",
    "md_round1and2_bothPS_df_its_q2 = q2.Metadata(md_round1and2_bothPS_df_its)\n",
    "\n",
    "# Generate distance matrices using 'all_dissts' utils\n",
    "rare_depth_its = 630\n",
    "dists_res_its = all_dists_no_tree(table_its_bothPS,\n",
    "                      rare_depth_its)\n",
    "\n",
    "# Generate ordinations (row=samples, cols=axes)\n",
    "pcoa_res_its = {}\n",
    "pcoa_res_its['Jaccard'] = pcoa(dists_res_its['Jaccard'].distance_matrix).pcoa.view(OrdinationResults).samples\n",
    "pcoa_res_its['RPCA'] = dists_res_its['RPCA'].biplot.view(OrdinationResults).samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpz7_toh5k/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpz7_toh5k/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpz7_toh5k/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total          3.0000     1.0000     \n",
      "Constrained    2.5312     0.8437    3\n",
      "Unconstrained  0.4688     0.1563    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.9447 0.8944 0.6922 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.30780 0.10561 0.05535 \n",
      "\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: /Users/Justin/Mycelium/ipynb/assets/stepwise-rda.R /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpy38bs2fa/ord_.tsv /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpy38bs2fa/mf_.txt /var/folders/jd/pqrf4k_j2ps8404x4rfp8_500000gn/T/tmpy38bs2fa/output.effect.size.tsv\n",
      "\n",
      "R version 4.0.3 (2020-10-10) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.5-7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call: rda(formula = Y_16S ~ 1, data = X_16S, scale = TRUE)\n",
      "\n",
      "              Inertia Rank\n",
      "Total               3     \n",
      "Unconstrained       3    3\n",
      "Inertia is correlations \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "PC1 PC2 PC3 \n",
      "  1   1   1 \n",
      "\n",
      "Call: rda(formula = Y_16S ~ round + sample_type + sample_type_2 +\n",
      "sample_type_3 + biomass_sample + extraction_kit_round, data = X_16S,\n",
      "scale = TRUE)\n",
      "\n",
      "              Inertia Proportion Rank\n",
      "Total          3.0000     1.0000     \n",
      "Constrained    2.3262     0.7754    3\n",
      "Unconstrained  0.6738     0.2246    3\n",
      "Inertia is correlations \n",
      "Some constraints were aliased because they were collinear (redundant)\n",
      "\n",
      "Eigenvalues for constrained axes:\n",
      "  RDA1   RDA2   RDA3 \n",
      "0.8449 0.7824 0.6989 \n",
      "\n",
      "Eigenvalues for unconstrained axes:\n",
      "    PC1     PC2     PC3 \n",
      "0.30111 0.21761 0.15508 \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>R2.adj</th>\n",
       "      <th>Df</th>\n",
       "      <th>AIC</th>\n",
       "      <th>F</th>\n",
       "      <th>Pr(&gt;F)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Jaccard</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.795789</td>\n",
       "      <td>59</td>\n",
       "      <td>-183.617659</td>\n",
       "      <td>33.231961</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.023947</td>\n",
       "      <td>6</td>\n",
       "      <td>-239.497438</td>\n",
       "      <td>10.498138</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">RPCA</th>\n",
       "      <th>+ sample_type_3</th>\n",
       "      <td>0.720444</td>\n",
       "      <td>59</td>\n",
       "      <td>-30.046851</td>\n",
       "      <td>22.315633</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>+ extraction_kit_round</th>\n",
       "      <td>0.020443</td>\n",
       "      <td>6</td>\n",
       "      <td>-62.068822</td>\n",
       "      <td>6.641175</td>\n",
       "      <td>0.0002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  R2.adj  Df         AIC          F  Pr(>F)\n",
       "Jaccard + sample_type_3         0.795789  59 -183.617659  33.231961  0.0002\n",
       "        + extraction_kit_round  0.023947   6 -239.497438  10.498138  0.0002\n",
       "RPCA    + sample_type_3         0.720444  59  -30.046851  22.315633  0.0002\n",
       "        + extraction_kit_round  0.020443   6  -62.068822   6.641175  0.0002"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform stepwise RDA-ANOVA\n",
    "es_all = {}\n",
    "use_ = ['round','sample_type','sample_type_2','sample_type_3','biomass_sample','extraction_kit_round']\n",
    "\n",
    "# Clean up meta (only stuff to run)\n",
    "mf_ord = md_round1and2_bothPS_df_its_q2.to_dataframe().copy()\n",
    "\n",
    "# Filter data\n",
    "keep_ = [v_ for v_ in mf_ord.columns\n",
    "         if len(set(mf_ord[v_])) > 1 and\n",
    "         len(set(mf_ord[v_])) < mf_ord.shape[0]//2]\n",
    "mf_ord = mf_ord[keep_]\n",
    "\n",
    "# Run stepwise ANOVA for all RDA ordinations\n",
    "for metric_, ord_ in  pcoa_res_its.items():\n",
    "    # get first three axes\n",
    "    ord_ = ord_[[0,1,2]]\n",
    "    ord_.columns = ['PC1','PC2','PC3']\n",
    "    # subset/match\n",
    "    mf_ord_ = mf_ord.copy()\n",
    "    shared_ids = list(set(ord_.index)\\\n",
    "                      & set(mf_ord_.index))\n",
    "    mf_ord_ = mf_ord_.loc[shared_ids,:]\n",
    "    ord_ = ord_.loc[shared_ids,:]\n",
    "    es_all[metric_] = run_stepwise_anova(ord_, mf_ord_, use_) #mf_ord_.columns)\n",
    "\n",
    "# Concat output from all runs and export\n",
    "es_alldf = pd.concat(es_all).rename({'+ sample_type_2':'Sample Type'}, axis=0)\n",
    "es_alldf.to_csv('/Users/Justin/Mycelium/UCSD/00_Knight_Lab/03_Extraction_test_12201/round_02/results/stepwise_anova/stepwise_anova_its.txt', sep='\\t')\n",
    "es_alldf\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (qiime2-amplicon-2024.10)",
   "language": "python",
   "name": "qiime2-amplicon-2024.10"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
